test/espnet2/asr/frontend/test_adapter_utils.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+
+import numpy
+import pytest
+import torch
+from fairseq.models.wav2vec.wav2vec2 import TransformerSentenceEncoderLayer
+
+from espnet2.asr.frontend.adapter_utils import *
+
+
+def test_add_adapters_wav2vec2():
```

Comment:
Does this code support other ssl models, e.g. hubert / wavlm? Another concern is that we need to download the wav2vec2 checkpoint if we use this. It will slow down the CI test a lot and may cause some issues if downloading fails. Can you change some configs to avoid it?

Reply:
Currently the code does not. I am certainly looking to support those! Do you think it it's better to do with currently wav2vec2 only or support those model within this PR? 

The current code gets around downloading the wav2vec2 checkpoint by onlying simulating with 3 `TransformerSentenceEncoderLayer` s and adding adapters to 2 of them. Does this work?

Reply:
Thanks for the information.
1. Can you please remind me if you support other models in your paper?
2. OK. If it skip downloading ckpt during, it is good. But another concern is how you ensure that it is compatible if fairseq updates the related implementation?

Reply:
1.We did also support HuBERT in our paper, do you prefer integrating that part code in this PR as well?
2.Yes, the problem about updated implementation is very real and it seems (from my view) that we could either do 
- pulling the wav2vec2.0 model and assume we don't know its internal implementation. This way I think we do not need to worry about fairseq updating implementation. But it's costly to pull the model and run them
- or current way which gets around the need of pulling the whole model. But this way assumes knowledge of  wav2vec's implementation and may fail if implementation are updated. 
which one do we prefer? Any other way of testing is greatly appreciated!